linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -353,6 +358,36 @@ private[emitter] final class KnowledgeGuardian(config: Emitter.Config) {
     private def computeSuperClass(linkedClass: LinkedClass): ClassName =
       linkedClass.superClass.fold[ClassName](null.asInstanceOf[ClassName])(_.name)
 
+    /** Computes the "version identity" of the fields of a `LinkedClass`.
+     *
+     *  The version identity is composed of
+     *
+     *  - The `version` of the `LinkedClass` itself, which will change every
+     *    time the definition of a field changes,
+     *  - A boolean indicating whether there is at least one `JSFieldDef`,
+     *    which will change very time the reachability analysis of the
+     *    `JSFieldDef`s changes (because we either keep all or none), and
+     *  - The list of names of the `FieldDef`s, which will change every time
+     *    the reachability analysis of the `FieldDef`s changes.
+     *
+     *  We do not try to use the names of `JSFieldDef`s because they are
+     *  `Tree`s, which are not efficiently comparable.
+     */
+    private def computeFieldDefsVersion(linkedClass: LinkedClass): FieldDefsVersion = {
+      val hasAnyJSField = linkedClass.fields.exists(_.isInstanceOf[JSFieldDef])
+      val hasAnyJSFieldVersion = Version.fromInt(if (hasAnyJSField) 1 else 0)
+      val version = Version.combine(linkedClass.version, hasAnyJSFieldVersion)
+
+      val scalaFieldNames = linkedClass.fields.collect {
+        case FieldDef(_, FieldIdent(name), _, _) => name
+      }
```

Comment:
Any reason we don't fold this into the overall version? Seems like:

- It would give simpler interface
- Wouldn't need custom code for comparison.
- Likely have a lower memory footprint (likely hardly matters TBH, but still).

We might need to add a `fromUTF8String` factory to `Version` for this, but that seems a like reasonable thing anyways.

Reply:
Ah indeed. It had not cross my mind, for some reason.

Updated.